crates/benchmark-runner/src/lib.rs

```diff
@@ -56,120 +16,51 @@ pub fn run_benchmark_ere<V>(
     zkvm_instance: V,
     action: Action,
     corpuses: &[BlocksAndWitnesses],
-) where
+) -> anyhow::Result<()>
+where
     V: zkVM + Sync,
 {
+    HardwareInfo::detect().to_path(format!(
```

Comment:
Save the `hardware.json` file in the root `zkevm-metrics` folder.

I still we need some extra work to make the output folder of the benchmark be more encapsulated, but we can do those improvements in future PRs. For now keeping the same style as before.